src/ucc/api/ucc.h

```diff
@@ -1574,5 +1588,229 @@ static inline ucc_status_t ucc_collective_test(ucc_coll_req_h request)
+} ucc_ee_params_t;
+
+/**
+ * @ingroup UCC_EVENT_DT
+ *
+ */
+typedef struct ucc_ee_attribs {
+    ucc_event_queue_h   post_queue;
+    ucc_event_queue_h   completion_queue;
+} ucc_ee_attribs_t;
+
+/**
+ * @ingroup UCC_EVENT
+ *
+ * @brief The routine creates the execution context for collective operations.
+ *
+ * @param [in] team     team handle
+ * @param [in] params   user provided params to customize the execution engine
+ * @param [out] ee      execution engine handle
+ *
+ * @parblock
+ *
+ * @b Description
+ *
+ * @endparblock
+ *
+ * @return Error code as defined by ucc_status_t
+ */
+ ucc_status_t ucc_ee_create(ucc_team_h team, const ucc_ee_params_t *params, ucc_ee_h *ee);
+
+ /**
+ * @ingroup UCC_EVENT
+ *
+ * @brief The routine destroys the execution context created for collective operations.
+ *
+ * @param [in] ee   Execution engine handle
+ *
+ * @parblock
+ *
+ * @b Description
+ *
+ * @endparblock
+ *
+ * @return Error code as defined by ucc_status_t
+ */
+ ucc_status_t ucc_ee_destroy(ucc_ee_h ee);
+
+/**
+ * @ingroup UCC_EVENT
+ *
+ * @brief The routine gets the event from the event queue.
+ *
+ * @param [in]  ee        execution engine handle
+ * @param [in]  eq_type        Event queue to pull the event
+ * @param [out] ev        Event structure fetched from the event queue
+ *
+ * @parblock
+ *
+ * @b Description
+ *
+ * @endparblock
+ *
+ * @return Error code as defined by ucc_status_t
+ */
+ucc_status_t ucc_ee_get_event(ucc_ee_h ee, ucc_event_queue_type_t eq_type,
+                ucc_ev_t *ev);
+
+
+/**
+ * @ingroup UCC_EVENT
+ *
+ * @brief The routine gets the event from the event queue.
+ *
+ * @param [in]  ee        execution engine handle
+ * @param [in]  eq_type   Event queue to pull the event
+ * @param [out] ev        Event structure fetched from the event queue
+ *
+ * @parblock
+ *
+ * @b Description
+ *
+ * @endparblock
+ *
+ * @return Error code as defined by ucc_status_t
+ */
+ucc_status_t ucc_ee_get_event(ucc_ee_h ee, ucc_event_queue_type_t eq_type,
+                ucc_ev_t *ev);
+
+/**
+ * @ingroup UCC_EVENT
+ *
+ * @brief The routine to set the event to the tail of the queue.
+ *
+ * @param [in]  ee        execution engine handle
+ * @param [in]  eq_type   Event queue to pull the event
+ * @param [in]  ev        Event structure fetched from the event queue
+ *
+ * @parblock
+ *
+ * @b Description
+ *
+ * @endparblock
+ *
+ * @return Error code as defined by ucc_status_t
+ */
+ucc_status_t ucc_ee_set_event(ucc_ee_h ee, ucc_event_queue_type_t eq_type,
+                ucc_ev_t ev);
+
+/**
+ * @ingroup UCC_EVENT
+ *
+ * @brief The routine polls the event queue and fetches the event from the head of the queue.
+ *
+ * @param [in]  ee        execution engine handle
+ * @param [in]  eq_type   Event queue to read the event
+ *
+ * @parblock
+ *
+ * @b Description
+ *
+ * @endparblock
+ *
+ * @return Error code as defined by ucc_status_t
+ */
+ucc_status_t ucc_ee_poll(ucc_ee_h ee, ucc_event_queue_type_t eq_type);
```

Comment:
should we have [out] param to get the event? can we combine this with ucc_ee_get_event() by returning head event if user dont specify event_type?